operator/pkg/apis/infrastructure/v1alpha1/controlplane.go

```diff
@@ -41,47 +42,39 @@ type ControlPlaneList struct {
 // master and etcd are configured to run. By default, KIT uses all the default
 // values and ControlPlaneSpec can be empty.
 type ControlPlaneSpec struct {
-	Infrastructure    `json:",inline"`
 	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
 	Master            MasterSpec `json:"master,omitempty"`
 	Etcd              ETCDSpec   `json:"etcd,omitempty"`
 }
 
-type Infrastructure struct {
-	VPCCidr        string   `json:"vpccidr,omitempty"`
-	Zones          []string `json:"zones,omitempty"`
-	PrivateSubnets []string `json:"privateSubnets,omitempty"`
-}
-
 // MasterSpec provides a way for the user to configure master instances and
 // custom flags for components running on master nodes like apiserver, KCM and
 // scheduler.
 type MasterSpec struct {
 	Instances         `json:",inline"`
-	Scheduler         *Config `json:"scheduler,omitempty"`
-	ControllerManager *Config `json:"controllerManager,omitempty"`
-	APIServer         *Config `json:"apiServer,omitempty"`
+	Scheduler         *Deployment `json:"scheduler,omitempty"`
```

Comment:
We also need other components like cert-controller and authenticator.

Reply:
yes will be adding them in the next iterations.

Reply:
We will likely also need a ...Deployment to enable arbitrary configuration. Though I suppose this could be customized outside of the Cluster CR.